apps/web/src/AppRoutes.tsx

```diff
@@ -99,10 +97,7 @@ export const AppRoutes = () => {
           <Route path=":identifier" element={<UpdateTenantPage />} />
         </Route>
         {isImprovedOnboardingEnabled ? (
-          <Route path={ROUTES.GET_STARTED} element={<GetStartedPage />}>
-            <Route path="" element={<GetStartedTab usecase={ProductUseCasesEnum.IN_APP} />} />
-            <Route path=":usecase" element={<GetStartedTab />} />
-          </Route>
+          <Route path={ROUTES.GET_STARTED} element={<GetStartedPage />} />
         ) : (
           <Route path={ROUTES.GET_STARTED} element={<GetStarted />} />
         )}
```

Comment:
This conditional (that requires a re-render) is why search params don't persist across refresh. Once we remove this, they will work as expected

Reply:
Not related to this scope
I wonder if we could/want to create a "loading" indication on our FF's so in such cases we would create a loading state, so we would avoid unnecessary rerenders and screen changes.

@antonjoel82 would love to hear your thoughts on it :)

Reply:
@djabarovgeorge What is `FF` in this context?

Sounds like a good UX improvement! We could do a temporary one, but overall my preference would be to upgrade our React version to use `Suspense` which will be a huge improvement overall.

Alternatively, one way I've done this in the past to avoid the re-routing issue is to route to a container page (just a page component that uses the hook to determine which content to show). Then, we'd delete that page later and point directly to the new functionality

Reply:
Sorry by i meant feature flag by FF.
Still had not the chance to use Suspense wonder how good is it.
Agree that a middleware component could help.

thanks for the feedback 🙃